service/init.go

```diff
@@ -139,9 +141,70 @@ func (smf *SMF) Initialize(c *cli.Context) error {
 		}()
 	}
 
+	if os.Getenv("MANAGED_BY_CONFIG_POD") == "true" {
+		logger.InitLog.Infoln("MANAGED_BY_CONFIG_POD is true")
+		go manageGrpcClient(factory.SmfConfig.Configuration.WebuiUri)
+	} else {
+		go func() {
+			logger.InitLog.Infoln("use helm chart config")
+			factory.ConfigPodTrigger <- true
+		}()
```

Comment:
It looks like the `else` statement should not be here (as in the "original" code).
```suggestion
```

When having the `else` statement, most of the tests fail (as you can see below)
```
        "Profile Name: profile4 , Profile Type: uetriggservicereq",
        "Ue's Passed: 1 , Ue's Failed: 4",
        "Profile Errors:",
        "imsi:imsi-208930100007530, procedure:REGISTRATION-PROCEDURE, error:triggering event:REGISTRATION-REQUEST-EVENT, expected event:AUTHENTICATION-REQUEST-EVENT, received event:REGISTRATION-REJECT-EVENT",
        "imsi:imsi-208930100007531, procedure:REGISTRATION-PROCEDURE, error:triggering event:REGISTRATION-REQUEST-EVENT, expected event:AUTHENTICATION-REQUEST-EVENT, received event:REGISTRATION-REJECT-EVENT",
        "imsi:imsi-208930100007532, procedure:REGISTRATION-PROCEDURE, error:triggering event:REGISTRATION-REQUEST-EVENT, expected event:AUTHENTICATION-REQUEST-EVENT, received event:REGISTRATION-REJECT-EVENT",
        "imsi:imsi-208930100007534, procedure:REGISTRATION-PROCEDURE, error:triggering event:REGISTRATION-REQUEST-EVENT, expected event:AUTHENTICATION-REQUEST-EVENT, received event:REGISTRATION-REJECT-EVENT",
        "Profile Status: FAIL",
        "Profile Name: profile2 , Profile Type: pdusessest",
        "Ue's Passed: 0 , Ue's Failed: 5",
        "Profile Errors:",
        "imsi:imsi-208930100007513, procedure:REGISTRATION-PROCEDURE, error:triggering event:REGISTRATION-REQUEST-EVENT, expected event:AUTHENTICATION-REQUEST-EVENT, received event:REGISTRATION-REJECT-EVENT",
        "imsi:imsi-208930100007514, procedure:REGISTRATION-PROCEDURE, error:triggering event:REGISTRATION-REQUEST-EVENT, expected event:AUTHENTICATION-REQUEST-EVENT, received event:REGISTRATION-REJECT-EVENT",
        "imsi:imsi-208930100007510, procedure:REGISTRATION-PROCEDURE, error:triggering event:REGISTRATION-REQUEST-EVENT, expected event:AUTHENTICATION-REQUEST-EVENT, received event:REGISTRATION-REJECT-EVENT",
        "imsi:imsi-208930100007512, procedure:REGISTRATION-PROCEDURE, error:triggering event:REGISTRATION-REQUEST-EVENT, expected event:AUTHENTICATION-REQUEST-EVENT, received event:REGISTRATION-REJECT-EVENT",
        "imsi:imsi-208930100007511, procedure:REGISTRATION-PROCEDURE, error:triggering event:REGISTRATION-REQUEST-EVENT, expected event:AUTHENTICATION-REQUEST-EVENT, received event:REGISTRATION-REJECT-EVENT",
        "Profile Status: FAIL",
        "Profile Name: profile7 , Profile Type: uereqpdusessrelease",
        "Ue's Passed: 0 , Ue's Failed: 5",
        "Profile Errors:",
        "imsi:imsi-208930100007561, procedure:REGISTRATION-PROCEDURE, error:triggering event:REGISTRATION-REQUEST-EVENT, expected event:AUTHENTICATION-REQUEST-EVENT, received event:REGISTRATION-REJECT-EVENT",
        "imsi:imsi-208930100007560, procedure:REGISTRATION-PROCEDURE, error:triggering event:REGISTRATION-REQUEST-EVENT, expected event:AUTHENTICATION-REQUEST-EVENT, received event:REGISTRATION-REJECT-EVENT",
        "imsi:imsi-208930100007562, procedure:REGISTRATION-PROCEDURE, error:triggering event:REGISTRATION-REQUEST-EVENT, expected event:AUTHENTICATION-REQUEST-EVENT, received event:REGISTRATION-REJECT-EVENT",
        "imsi:imsi-208930100007563, procedure:REGISTRATION-PROCEDURE, error:triggering event:REGISTRATION-REQUEST-EVENT, expected event:AUTHENTICATION-REQUEST-EVENT, received event:REGISTRATION-REJECT-EVENT",
        "imsi:imsi-208930100007564, procedure:REGISTRATION-PROCEDURE, error:triggering event:REGISTRATION-REQUEST-EVENT, expected event:AUTHENTICATION-REQUEST-EVENT, received event:REGISTRATION-REJECT-EVENT",
        "Profile Status: FAIL",
        "Profile Name: profile1 , Profile Type: register",
        "Ue's Passed: 0 , Ue's Failed: 5",
        "Profile Errors:",
        "imsi:imsi-208930100007501, procedure:REGISTRATION-PROCEDURE, error:triggering event:REGISTRATION-REQUEST-EVENT, expected event:AUTHENTICATION-REQUEST-EVENT, received event:REGISTRATION-REJECT-EVENT",
        "imsi:imsi-208930100007504, procedure:REGISTRATION-PROCEDURE, error:triggering event:REGISTRATION-REQUEST-EVENT, expected event:AUTHENTICATION-REQUEST-EVENT, received event:REGISTRATION-REJECT-EVENT",
        "imsi:imsi-208930100007500, procedure:REGISTRATION-PROCEDURE, error:triggering event:REGISTRATION-REQUEST-EVENT, expected event:AUTHENTICATION-REQUEST-EVENT, received event:REGISTRATION-REJECT-EVENT",
        "imsi:imsi-208930100007503, procedure:REGISTRATION-PROCEDURE, error:triggering event:REGISTRATION-REQUEST-EVENT, expected event:AUTHENTICATION-REQUEST-EVENT, received event:REGISTRATION-REJECT-EVENT",
        "imsi:imsi-208930100007502, procedure:REGISTRATION-PROCEDURE, error:triggering event:REGISTRATION-REQUEST-EVENT, expected event:AUTHENTICATION-REQUEST-EVENT, received event:REGISTRATION-REJECT-EVENT",
        "Profile Status: FAIL",
        "Profile Name: profile3 , Profile Type: anrelease",
        "Ue's Passed: 0 , Ue's Failed: 5",
        "Profile Errors:",
        "imsi:imsi-208930100007521, procedure:REGISTRATION-PROCEDURE, error:triggering event:REGISTRATION-REQUEST-EVENT, expected event:AUTHENTICATION-REQUEST-EVENT, received event:REGISTRATION-REJECT-EVENT",
        "imsi:imsi-208930100007524, procedure:REGISTRATION-PROCEDURE, error:triggering event:REGISTRATION-REQUEST-EVENT, expected event:AUTHENTICATION-REQUEST-EVENT, received event:REGISTRATION-REJECT-EVENT",
        "imsi:imsi-208930100007520, procedure:REGISTRATION-PROCEDURE, error:triggering event:REGISTRATION-REQUEST-EVENT, expected event:AUTHENTICATION-REQUEST-EVENT, received event:REGISTRATION-REJECT-EVENT",
        "imsi:imsi-208930100007522, procedure:REGISTRATION-PROCEDURE, error:triggering event:REGISTRATION-REQUEST-EVENT, expected event:AUTHENTICATION-REQUEST-EVENT, received event:REGISTRATION-REJECT-EVENT",
        "imsi:imsi-208930100007523, procedure:REGISTRATION-PROCEDURE, error:triggering event:REGISTRATION-REQUEST-EVENT, expected event:AUTHENTICATION-REQUEST-EVENT, received event:REGISTRATION-REJECT-EVENT",
        "Profile Status: FAIL",
        "Profile Name: profile5 , Profile Type: deregister",
        "Ue's Passed: 1 , Ue's Failed: 4",
        "Profile Errors:",
        "imsi:imsi-208930100007544, procedure:REGISTRATION-PROCEDURE, error:triggering event:REGISTRATION-REQUEST-EVENT, expected event:AUTHENTICATION-REQUEST-EVENT, received event:REGISTRATION-REJECT-EVENT",
        "imsi:imsi-208930100007540, procedure:REGISTRATION-PROCEDURE, error:triggering event:REGISTRATION-REQUEST-EVENT, expected event:AUTHENTICATION-REQUEST-EVENT, received event:REGISTRATION-REJECT-EVENT",
        "imsi:imsi-208930100007543, procedure:REGISTRATION-PROCEDURE, error:triggering event:REGISTRATION-REQUEST-EVENT, expected event:AUTHENTICATION-REQUEST-EVENT, received event:REGISTRATION-REJECT-EVENT",
        "imsi:imsi-208930100007541, procedure:REGISTRATION-PROCEDURE, error:triggering event:REGISTRATION-REQUEST-EVENT, expected event:AUTHENTICATION-REQUEST-EVENT, received event:REGISTRATION-REJECT-EVENT",
        "Profile Status: FAIL"
```

When I remove the `else` statement (same as in the original code), all tests pass es expected. Did you have a chance to run E2E tests?
```
        "Profile Name: profile1 , Profile Type: register",
        "Ue's Passed: 5 , Ue's Failed: 0",
        "Profile Status: PASS",
        "Profile Name: profile2 , Profile Type: pdusessest",
        "Ue's Passed: 5 , Ue's Failed: 0",
        "Profile Status: PASS",
        "Profile Name: profile3 , Profile Type: anrelease",
        "Ue's Passed: 5 , Ue's Failed: 0",
        "Profile Status: PASS",
        "Profile Name: profile7 , Profile Type: uereqpdusessrelease",
        "Ue's Passed: 5 , Ue's Failed: 0",
        "Profile Status: PASS",
        "Profile Name: profile5 , Profile Type: deregister",
        "Ue's Passed: 5 , Ue's Failed: 0",
        "Profile Status: PASS",
        "Profile Name: profile4 , Profile Type: uetriggservicereq",
        "Ue's Passed: 5 , Ue's Failed: 0",
        "Profile Status: PASS"
```

Reply:
I am not able to test else option, but if clause works for me. Anyway, I removed the else clause.